test/unit/org/apache/cassandra/tools/ToolRunner.java

```diff
@@ -304,17 +248,12 @@ public boolean isRunning()
     }
 
     public boolean waitFor()
-    {
-        return waitFor(defaultTimeoutMillis, TimeUnit.MILLISECONDS);
-    }
-
-    public boolean waitFor(long time, TimeUnit timeUnit)
```

Comment:
Why kill this method? and if it is so why not remove `defaultTimeoutMillis` along?

Reply:
It causes subtle bugs since the API doesn't handle the case the process has not completed.

The APIs check for exit code and will call waitFor, but if waitFor times out then the exit code stays 0 and the test passes with a bad result.  Rather than trying to change the API to spilt these two cases, I just removed the timeout waitFor since I couldn't find a single usage which properly handled timeout; all usages assumed no timeout and would move forward incorrectly when a timeout happens.

Reply:
removed default timeout

Reply:
Ok I see you did a first pass on CASSANDRA-15991 where it is being used. Let's remove it for the time now...